packages/backend/src/paymentPointerKey/service.ts

```diff
@@ -0,0 +1,91 @@
+import { TransactionOrKnex } from 'objection'
+
+import { PaymentPointerKey } from './model'
+import { BaseService } from '../shared/baseService'
+import { JWKWithRequired } from 'auth'
+
+export interface PaymentPointerKeyService {
+  create(options: CreateOptions): Promise<PaymentPointerKey>
+  revoke(id: string): Promise<PaymentPointerKey | undefined>
+  getKeysByPaymentPointerId(
+    paymentPointerId: string
+  ): Promise<PaymentPointerKey[]>
+}
+
+interface ServiceDependencies extends BaseService {
+  knex: TransactionOrKnex
+}
+
+export async function createPaymentPointerKeyService({
+  logger,
+  knex
+}: ServiceDependencies): Promise<PaymentPointerKeyService> {
+  const log = logger.child({
+    service: 'PaymentPointerKeyService'
+  })
+  const deps: ServiceDependencies = {
+    logger: log,
+    knex
+  }
+  return {
+    create: (options) => create(deps, options),
+    revoke: (id) => revoke(deps, id),
+    getKeysByPaymentPointerId: (paymentPointerId) =>
+      getKeysByPaymentPointerId(deps, paymentPointerId)
+  }
+}
+
+interface CreateOptions {
+  paymentPointerId: string
+  jwk: JWKWithRequired
+}
+
+async function create(
+  deps: ServiceDependencies,
+  options: CreateOptions
+): Promise<PaymentPointerKey> {
+  const key = await PaymentPointerKey.query(deps.knex).insertAndFetch({
+    paymentPointerId: options.paymentPointerId,
+    jwk: options.jwk
+  })
+  return key
+}
+
+async function revoke(
+  deps: ServiceDependencies,
+  id: string
+): Promise<PaymentPointerKey | undefined> {
```

Comment:
I'm good with this as is, but another option would be to return `PaymentPointerKeyError` for unknown or already revoked keys.
See:
https://github.com/interledger/rafiki/blob/603df09615a2f458227baf3e23f99b3a0b103921/packages/backend/src/open_payments/payment/incoming/service.ts#L285-L308

Reply:
I think this could be implemented when we have a few more error cases.